_drafts/floating-point-101.md

```diff
@@ -0,0 +1,444 @@
+to `0`… but that implicit leading `1` is standing in the way.
+
+Here's the trick: when the significand is `0`, setting the biased exponent to `0` makes the significand's leading digit
+_also_ `0`. _Voilà_, `0` as a result! That's a useful number to have.
+
+Hmm, what if we set the _sign_ to `1` at the same time? That signifies a negative value, but it's obviously ridiculous
+for _zero_ to be neg– WHAT?! According to all sources _(what sources now)_ we _do_ actually get `-0` this way. It's not
+even as absurd as it seems at first glance: for practically all intents and purposes `-0 == +0` and this only doesn't
+hold when there's an important reason – we'll go into it in [The Enormous](#the-enormous).
+
+> See how the signed zero is stored in binary by trying `-0` in [the calculator](#calculator).
+
+## The Undefined
+
+Some mathematical expressions simply cannot be evaluated. Take `0 / 0` for instance: the result of this is said to be
+_undefined_ in mathematics. This means there is no answer at all (not to be confused with JavaScript's `undefined`
+value, which means the value hasn't been initiated) or, in other words, the answer certainly is _not a number_.
+
+Programs used to crash immediately whenever they encountered such expressions and in fact they still do when this
+happens with integers – integer formats don't have a way of safely representing an undefined value, so there's no
+recourse. Floating-point has a solution though: the **NaN** value, short for "not a number".
+
+> See how NaN is stored in binary by trying `NaN` in [the calculator](#calculator).
+
+Specifically, you get a NaN whenever:
+
+1. An [indeterminate form](https://en.wikipedia.org/wiki/Indeterminate_form) is encountered:
+
+    - `0 / 0`
+    - `infinity / infinity`
+    - `0 * infinity`
+    - `infinity ± infinity`
+    - `x % 0`
+    - `infinity % x`
+
+2. The result of an operation would have to be a complex number (floating-point only represents reals):
+    - `sqrt(x)` for `x` below 0
+    - `log(x)` for `x` below 0
+    - `asin(x)` and `acos(x)` for `x` below -1 or above 1
+
+Additionally, NaNs propagate, so a NaN _anywhere_ in an expression makes it almost guaranteed that the final result will
+be NaN too.
+
+Be careful with comparisons, as NaNs uniquely have an **unordered** relation to all values, instead of being smaller,
+greater, or equal. That means `<`, `>`, `<=`, `>=`, and `==` comparisons involving a NaN _always_ come out false – no
+matter if the special value is on the left or right side, or even if it's on _both_ (so a NaN is never ever equal to
+another NaN). On the other hand `!=` always comes out true. An important implication of the above is that `>=` _cannot_
+be implemented as just a negation of `<` for floating-point!
+
+## The Microscopic
+
+When a result of a calculation is so tiny that it can't be represented by a normal number we see **underflow** occur.
```

Comment:
I'd re-use the distance analogy from above to drive this point home :)

Reply:
Hmm, I acknowledge the idea, but I'm a bit too lazy for it RN 😅